src/services/bundle.service.js

```diff
@@ -0,0 +1,149 @@
+    } else {
+      entry.response.outcome = result.data;
+    }
+    entries.push(entry);
+  });
+
+  bundle.entry = entries;
+  return bundle;
+};
+
+/**
+ * Uploads bundle to server.
+ * @param {Object} request the request object passed in by the user
+ * @param {Object} reply the response object
+ *
+ * @returns {Object} transaction-response/batch-response bundle
+ */
+const uploadTransactionOrBatchBundle = async (request, reply) => {
+  request.log.info('Base >>> Transaction/Batch Bundle Upload');
+  const { resourceType, type, entry: entries } = request.body;
+
+  if (resourceType !== 'Bundle') {
+    reply.code(400).send(new Error(`Expected 'resourceType: Bundle', but received 'resourceType: ${resourceType}'.`));
+  }
+  if (!['transaction', 'batch'].includes(type.toLowerCase())) {
+    reply.code(400).send(new Error(`Expected 'type: transaction' or 'type: batch'. Received 'type: ${type}'.`));
+  }
+
+  if (type.toLowerCase() == 'transaction') {
+    const requestResults = await uploadResourcesFromTxnBundle(entries, reply);
+    const bundle = createResponseBundle(requestResults, reply, 'transaction');
+    request.log.info('Transaction bundle successfully uploaded to server');
+    return bundle;
+  } else {
+    const requestResults = await uploadResourcesFromBatchBundle(entries);
+    const bundle = createResponseBundle(requestResults, reply, 'batch');
+    request.log.info('Batch bundle successfully uploaded to server');
+    return bundle;
+  }
+};
+
+/**
+ * Scrubs transaction bundle entries and uploads each entry to the server.
+ * @param {Array} entries entries from POSTed transaction bundle
+ * @param {Object} reply the response object
+ * @returns array of request results
+ */
+const uploadResourcesFromTxnBundle = async (entries, reply) => {
+  const scrubbedEntries = replaceReferences(entries);
+  const requestsArray = scrubbedEntries.map(async entry => {
+    const { method } = entry.request;
+    return insertBundleResources(entry, method).catch(e => {
+      // send 400 error and stop any further operations from occurring
+      reply.code(400).send(e.message);
+    });
+  });
+  const requestResults = await Promise.all(requestsArray);
+  return requestResults;
+};
+
+/**
+ * Scrubs batch bundle entries and uploads each entry to the server.
+ * @param {Array} entries entries from POSTed batch bundle
+ * @returns array of request results
+ */
+const uploadResourcesFromBatchBundle = async entries => {
+  const scrubbedEntries = replaceReferences(entries);
+  const requestsArray = scrubbedEntries.map(async entry => {
+    const { method } = entry.request;
+    return insertBundleResources(entry, method).catch(() => {
+      // exclude results for resource that produced an error, but
+      // continue executing remaining operations
+      return null;
+    });
+  });
+  const requestResults = await Promise.all(requestsArray);
+  return requestResults.filter(results => results !== null);
+};
```

Comment:
It really doesn't matter, but if you want to condense this a bit I think you can combine them and just have the stuff that is different in an if else statement? I only suggest this because there's a bit of overlap but it's all functional so doesn't matter :)

Reply:
Good point, I kept these separate because I was in a bit of a time crunch, but I agree we can definitely simplify this